nixos/doc/manual/development/freeform-modules.xml

```diff
@@ -0,0 +1,69 @@
+<section xmlns="http://docbook.org/ns/docbook"
+        xmlns:xlink="http://www.w3.org/1999/xlink"
+        xmlns:xi="http://www.w3.org/2001/XInclude"
+        version="5.0"
+        xml:id="sec-freeform-modules">
+ <title>Freeform modules</title>
+ <para>
+  Freeform modules allow you to define values for option paths that have not been declared explicitly. This can be used to add attribute-specific types to <literal>attrsOf</literal> options.
+ </para>
+ <para>
+  This feature can be enabled by using the option <literal>_module.freeformType</literal> to define a freeform type. By doing this, all assigments without an associated option will get merged using the freeform type and combined into the resulting <literal>config</literal> set. Since this feature influences the whole module evaluation, it is mostly recommended for submodule options only.
```

Comment:
```suggestion
  This feature can be enabled by using the option <literal>_module.freeformType</literal> to define a freeform type. By doing this, all assigments without an associated option will be merged using the freeform type and combined into the resulting <literal>config</literal> set. Since this feature nullifies name checking for entire option trees, it is only recommended for use in submodules.
```